Formula/k/kubectl-cnpg.rb

```diff
@@ -0,0 +1,32 @@
+class KubectlCnpg < Formula
+  desc "CloudNativePG plugin for kubectl"
+  homepage "https://cloudnative-pg.io/"
+  url "https://github.com/cloudnative-pg/cloudnative-pg/archive/refs/tags/v1.23.1.tar.gz"
+  sha256 "c13373baf2a0d177d82f85be66b963ff6d55853aaafc7cbbe9afa904d75b8695"
+  license "Apache-2.0"
+  head "https://github.com/cloudnative-pg/cloudnative-pg.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildVersion=#{version}
+      -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildCommit=#{tap.user}
+      -X github.com/cloudnative-pg/cloudnative-pg/pkg/versions.buildDate=#{time.iso8601}
+    ]
+    system "go", "build", *std_go_args(ldflags:, output: "#{bin}/kubectl-cnpg"), "./cmd/kubectl-cnpg"
+    generate_completions_from_executable(bin/"kubectl-cnpg", "completion")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/kubectl-cnpg version")
+
+    expected_error = Regexp.new(<<~EOS)
+      Error: failed to get API group resources: unable to retrieve the complete list of server APIs: \
+      postgresql.cnpg.io/v1: Get "http://localhost:8080/apis/postgresql.cnpg.io/v1": \
+      dial tcp ((\\[::1\\])|(127\\.0\\.0\\.1)):8080: connect: connection refused
+    EOS
```

Comment:
Could we simplify this regex a bit so that it's less likely to be broken in case upstream decides to change the form of the error? One way to do that is to just check for matches against fragments of this string (e.g. `Error:`, `connection refused`).

Also, I don't think you actually need a regex here, you can just convert this into a string if you want to keep the whole message.

Reply:
The funny thing is that I already had that as a normal string, but the macOS CI build failed. On my machine (Linux) and in the Linux CI it's an IPv4 error string and on macOS it's an IPv6 error string.

I'll simplify it and just use a string contains 👍

Reply:
Fixed